lib/controllers/BookmarkController.dart

```diff
@@ -1,33 +1,82 @@
+import 'package:bookmarks/abstractions/Controller.dart';
 import 'package:bookmarks/models/Bookmark.dart';
+import 'package:bookmarks/models/Folder.dart';
 import 'package:bookmarks/models/User.dart';
 import 'package:bookmarks/services/BookmarkService.dart';
-import 'package:bookmarks/widgets/HomeWidget.dart';
+import 'package:bookmarks/services/FolderService.dart';
 import 'package:bookmarks/views/BookmarkListView.dart';
-import 'package:bookmarks/abstractions/Controller.dart';
+import 'package:bookmarks/widgets/HomeWidget.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:url_launcher/url_launcher.dart';
 
 class BookmarkController extends Controller<HomeWidget> {
   int _counter = 0;
 
+  int _folderId = -1;
+
   List<Bookmark> _bookmarks;
+  List<Folder> _folders;
 
   int get counter => _counter;
+
+  int get folderId => _folderId;
+
   List<Bookmark> get bookmarks => _bookmarks;
 
+  Folder get folder =>
+      this._folderId == -1 ? null : findFolder(this._folders, this._folderId);
```

Comment:
It would be nice to make clearer what the edge case and what the regular case is (and I might not like ternary operators too much ^^) How about changing this to
```suggestion
  Folder get folder {
    if (this._folderId == -1) return null;

    return findFolder(this._folders, this._folderId);
  }
```